src/support.cpp

```diff
@@ -240,10 +244,16 @@ void AreaSupport::generateSupportAreas(SliceDataStorage& storage, unsigned int m
         AreaSupport::detectOverhangPoints(storage, mesh, overhang_points, layer_count, supportMinAreaSqrt);
     }
 
-    std::deque<std::pair<Polygons, Polygons>> basic_and_full_overhang_above;
-    for (unsigned int layer_idx = support_layer_count - 1; layer_idx != support_layer_count - 1 - layerZdistanceTop ; layer_idx--)
+    std::vector<Polygons> basic_overhang_per_layer;
+    std::vector<Polygons> full_overhang_per_layer;
+    basic_overhang_per_layer.resize(support_layer_count);
+    full_overhang_per_layer.resize(support_layer_count);
+#pragma omp parallel for default(none) shared(basic_overhang_per_layer, full_overhang_per_layer, support_layer_count, storage, mesh, max_dist_from_lower_layer) schedule(dynamic)
+    for (unsigned int layer_idx = 0; layer_idx < support_layer_count; layer_idx++)
     {
-        basic_and_full_overhang_above.push_front(computeBasicAndFullOverhang(storage, mesh, layer_idx, max_dist_from_lower_layer));
+        std::pair<Polygons, Polygons> basic_and_full_overhang = computeBasicAndFullOverhang(storage, mesh, layer_idx, max_dist_from_lower_layer);
+        basic_overhang_per_layer[layer_idx] = basic_and_full_overhang.first;
+        full_overhang_per_layer[layer_idx] = basic_and_full_overhang.second;
```

Comment:
You seriously find a vector of pairs better than two vectors?! Why?

In my eyes a better question would be: why was it a queue of pairs in the first place? Because we threw away old data from both at the same time.

Now we don't throw any data away, so that we can compute it all at the same time (at the cost of extra memory usage)